### ISSUE TITLE
Unused Variable for import used in object literal property value shorthand

### ISSUE THREAD

**Issue by unional** [palantir/tslint#1005]
**TypeScript Version:** 
1.8.2

**Code**

```ts
import funcA from './funcA'; // Unused Variable 'funcA'

// either
export default {
  funcA
}

// or
let a = {
  funcA
};
```

It is flagged by tslint through language service as a warning:
![image](https://cloud.githubusercontent.com/assets/3254987/13404129/248b84ec-decd-11e5-972e-4fe36eb028c8.png)
![image](https://cloud.githubusercontent.com/assets/3254987/13404189/5bd65b66-decd-11e5-9726-b4cc2bd2642b.png)

The warning is gone when shorthand is not used:
![image](https://cloud.githubusercontent.com/assets/3254987/13404204/6f3d8fda-decd-11e5-8fd9-04866d76f85b.png)

**Comment by JKillian** [comment]
@unional This looks like the same issue as #684. Perhaps your IDE is using an older version of TypeScript?

**Comment by unional** [comment]
Using 1.8.2

**Comment by unional** [comment]
tslint 3.5.0

**Comment by JKillian** [comment]
@unional I tested TSLint out on the exact code you provided with the versions you specified and I couldn't duplicate the error. My guess is that internally somewhere, you're using a version of TS that's older than 1.8.2.

**Comment by unional** [comment]
I use a local copy of TypeScript to make sure it i running with the correct TS version:
```sh
E:\hwong\p4\projects\pandash
λ npm list typescript
pandash@0.1.5 E:\hwong\p4\projects\pandash
└── typescript@1.8.2

E:\hwong\p4\projects\pandash
λ npm list -g tslint
C:\Users\hwong\AppData\Roaming\npm
└── tslint@3.5.0
```

```js
// .vscode/settings.json
{
  "typescript.tsdk": "node_modules/typescript/lib"
}
```

It *should* be using the right TS version.

**Comment by unional** [comment]
Just tried with `typescript@next` (i.e. `1.9.0-dev.20160229`), the problem still presist.

**Comment by adidahiya** [comment]
Are you verifying this bug via VSCode or the `tslint` CLI? Please make sure to try the latter.

**Comment by unional** [comment]
CLI is working. Opened issue in `vscode-tslint`. Closing this. 

Thanks!

**Comment by unional** [comment]
Re-open, it is actually not working. There were problem with my gulp config.
```sh
[15:19:51] [gulp-tslint] error (no-unused-variable) array.default.ts[1, 8]: unused variable: 'composeMap'
[15:19:51] [gulp-tslint] error (no-unused-variable) array.default.ts[2, 8]: unused variable: 'composeReduce'
[15:19:51] [gulp-tslint] error (no-unused-variable) array.default.ts[3, 8]: unused variable: 'composeSequential'
```
![image](https://cloud.githubusercontent.com/assets/3254987/13479275/9297aaf6-e08a-11e5-9567-28545bde4406.png)

**Comment by JKillian** [comment]
Hi @unional, I tested your code with the newest version of TSLint and TypeScript and didn't get any of these errors. It seems likely to me that there's an old version of TypeScript hidden somewhere in your workflow that is causing all this.

As a sidenote, you probably don't want to use the `.ts` extension when importing fields form modules! Something like `import composeMap from './array/composeMap';` is more correct.

**Comment by adidahiya** [comment]
re: @JKillian's side note: when compiled, that code will break because the ".ts" extension remains in built JS file.

**Comment by unional** [comment]
> As a sidenote, you probably don't want to use the .ts extension when importing fields form modules! 

Yes.  I understanding that.  Thanks for pointing it out. I'm trying to figure out the proper workflow of authoring package in typescript in `jspm` environment.
https://github.com/jspm/jspm-cli/issues/1545
Maybe "module: system" would make that work, but very limiting. :disappointed: 

Regarding the error, I don't know why it happens to me and not on yours.  Will try to see if there are pieces of the puzzle missing.

**Comment by JKillian** [comment]
Sounds good, let us know if you figure it out. If other people have this same issue, I'll reopen this in the future.
